HotSpot/View Controllers /CurrentAndPastDetails.m

```diff
@@ -68,9 +75,36 @@ - (void)prepareForSegue:(UIStoryboardSegue *)segue sender:(id)sender {
     }
 }
 
+
+
 - (IBAction)bookingBackPressed:(id)sender {
     [self dismissViewControllerAnimated:YES completion:nil];
 }
 
+- (IBAction)directionsPressed:(id)sender {
+    [self openMap]; 
+}
+
+
+- (void) openMap{
+    Listing *listing = self.booking.listing;
+    [listing fetchInBackgroundWithBlock:^(PFObject * _Nullable object, NSError * _Nullable error){
+        CLLocationDegrees longitude = listing.address.longitude;
+        CLLocationDegrees latitude= listing.address.latitude;
+        NSString* directionsURL = [NSString stringWithFormat:@"http://maps.apple.com/?saddr=%f,%f&daddr=%f,%f", self.locationManager.location.coordinate.latitude, self.locationManager.location.coordinate.longitude,  latitude, longitude];
+        if ([[UIApplication sharedApplication] respondsToSelector:@selector(openURL:options:completionHandler:)]) {
+            [[UIApplication sharedApplication] openURL:[NSURL URLWithString: directionsURL] options:@{} completionHandler:^(BOOL success) {}];
+        } else {
+            
+            [[UIApplication sharedApplication] openURL:[NSURL URLWithString:directionsURL] options:@{MKLaunchOptionsDirectionsModeKey: MKLaunchOptionsDirectionsModeDriving} completionHandler:^(BOOL success) {
```

Comment:
whoa, not sure if you should do that! you're basically saying, if UIApplication contains this method, call it, else, call it anyway. That could lead to a crash!